integrations/lib/multiwoven/integrations/source/athena/client.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require "aws-sdk-athena"
+
+module Multiwoven::Integrations::Source
+  module AWSAthena
+    include Multiwoven::Integrations::Core
+    class Client < SourceConnector
+      def check_connection(connection_config)
+        connection_config = connection_config.with_indifferent_access
+        create_connection(connection_config)
+        ConnectionStatus.new(status: ConnectionStatusType["succeeded"]).to_multiwoven_message
+      rescue StandardError => e
+        ConnectionStatus.new(status: ConnectionStatusType["failed"], message: e.message).to_multiwoven_message
+      end
+
+      def discover(connection_config)
+        connection_config = connection_config.with_indifferent_access
+        query = "SELECT table_name, column_name, data_type, is_nullable FROM information_schema.columns WHERE table_schema = '#{connection_config[:schema]}' ORDER BY table_name, ordinal_position;"
+
+        db = create_connection(connection_config)
+        response = db.start_query_execution(
+          query_string: query,
+          result_configuration: { output_location: connection_config[:output_location] }
+        )
+        query_execution_id = response[:query_execution_id]
+        # Polling for query execution completion
```

Comment:
The discover and read query fetch is being done asynchronously. How will we get a response for each call? Can we do it synchronously?